### ISSUE TITLE
Training code release and experiment reproducibility 

### ISSUE THREAD

**Issue by FrancescoFornasa** [YvanYin/Metric3D#53]
Dear Sirs,

I am trying to fully replicate your experiments in order to validate the claims of your paper.
There are, however, many points that seem to have been explained in far too little details to be implemented.

As an example, in both the paper and the supplementary material, you claim to have used the PWNL loss as implemented in LeReS. This, however, seems improbable at least for the Lidar datasets as their depth is sparse, rendering it impossible to compute the edges needed by PWNL.

As the things are now, it is impossible to validate the claims you have made by replicating your training experiment. This makes your amazing work impossible to use for research purposes.

It would be of course best if you could disclose the training pipeline but it seems you are having IP issues since last summer. In the meantime, can you please at least explain thoroughly how you have used the PWNL loss? 

- Have you used both the edges and planar components?

- Have you used PWNL also for lidar datasets? If so, how?

Thank you for your amazing work and kind response

**Comment by FrancesC0de** [comment]
I'm also interested about PWNL loss application, following

**Comment by leonardo-fernandesoliveira** [comment]
Also looking forward to the release of the training code.

**Comment by YvanYin** [comment]
We will release the training code very soon.

**Comment by YvanYin** [comment]
> Dear Sirs,
> 
> I am trying to fully replicate your experiments in order to validate the claims of your paper. There are, however, many points that seem to have been explained in far too little details to be implemented.
> 
> As an example, in both the paper and the supplementary material, you claim to have used the PWNL loss as implemented in LeReS. This, however, seems improbable at least for the Lidar datasets as their depth is sparse, rendering it impossible to compute the edges needed by PWNL.
> 
> As the things are now, it is impossible to validate the claims you have made by replicating your training experiment. This makes your amazing work impossible to use for research purposes.
> 
> It would be of course best if you could disclose the training pipeline but it seems you are having IP issues since last summer. In the meantime, can you please at least explain thoroughly how you have used the PWNL loss?
> 
> * Have you used both the edges and planar components?
> * Have you used PWNL also for lidar datasets? If so, how?
> 
> Thank you for your amazing work and kind response

We do not use PWNL for lidar datasets. As lidars are too sparse, we cannot obtain accurate plane information. We only use it on some dense datasets. Edges and planar components are both applied in our metric3dv1.

**Comment by YvanYin** [comment]
Training code will be released this week.
